### DISCUSSION TITLE
How to enable NAT64 capabilities on Border Router?

### DISCUSSION THREAD

**Discussion by lecontr2** [openthread/openthread#9389] [Q&A] [answered]
Hello everyone, I've been reviewing the documentation posted [here](https://openthread.io/codelabs/openthread-border-router-nat64#2) but I'm wondering if it is possible to enable the NAT64 capabilities from a Thread node client? 

I'm trying to test this feature on publicly released OTBRs (Eero, Nest, Aeotec, HomePod), and I can see that NAT64 is allowed when requesting the DNS configuration from the Border Router but I seem to be unable to obtain a NAT64 prefix from any of them with the exception of Apple's HomePod Mini, preventing me from being able to access the internet from my Thread node.

**Comment by gabekassel** [comment, marked as answer]
eero provides DNS64/NAT64. If you build your OpenThread device with SRP client and DNS client you can test this.

**Reply by lecontr2** [reply]
Oh incredible, I can see the NAT64 prefix now. Regarding the DNS Queries, it seems like it's unable to obtain IPv6 or IPv4 addresses for the requests I make. Do you have any suggestions as to what I should look for?

**Reply by gabekassel** [reply]
If using the cli you need to do something like:

- srp autostart enable
- dns config (should see an fd5d: prefix)
- dns resolve ipv4.google.com
- will return an fd5d prefixed address that synthesizes thev4 address
